src/arch/linux/mapping/mapfile.c

```diff
@@ -93,20 +93,19 @@ static int commit(void *ptr, size_t size)
 
 static int open_mapping_f(int cap)
 {
-    int mapsize, estsize, padsize;
+    int mapsize = 0;
+    int estsize, padsize;
 
     if (cap) Q_printf("MAPPING: open, cap=%s\n",
 	  decode_mapping_cap(cap));
 
     padsize = 4*1024;
 
     /* first estimate the needed size of the mapfile */
-    mapsize  = HMASIZE >> 10;	/* HMA */
- 				/* VGAEMU */
     mapsize += config.vgaemu_memsize;
     mapsize += config.ems_size;	/* EMS */
     mapsize += config.xms_size;	/* XMS */
-    mapsize += LOWMEM_SIZE >> 10; /* Low Mem */
+    mapsize += config.dpmi_base >> 10; /* Low Mem up to dpmi */
```

Comment:
I'd change this to
```
mapsize += config.ext_mem;
mapsize += (LOWMEM_SIZE + HMASIZE) >> 10; /* Low Mem + HMA */
```
then limit `config.xms_map_size` to `_min(config.xms_size << 10, config.dpmi_base - (LOWMEM_SIZE + HMASIZE + EXTMEM_SIZE)`